src/query/sql/src/planner/binder/copy.rs

```diff
@@ -464,6 +495,148 @@ impl<'a> Binder {
         })))
     }
 
+    /// Bind COPY INTO <table> FROM <query>
+    async fn bind_copy_from_query_into_table(
+        &mut self,
+        bind_context: &BindContext,
+        stmt: &CopyStmt,
+        src_query: &Query,
+        dst_catalog_name: &str,
+        dst_database_name: &str,
+        dst_table_name: &str,
+    ) -> Result<Plan> {
+        // Validation mode.
+        let validation_mode = ValidationMode::from_str(stmt.validation_mode.as_str())
+            .map_err(ErrorCode::SyntaxException)?;
+
+        // dst
+        let dst_table = self
+            .ctx
+            .get_table(dst_catalog_name, dst_database_name, dst_table_name)
+            .await?;
+
+        // src
+        let (select_list, location, alias) = check_transform_query(src_query)?;
+        if matches!(location, FileLocation::Uri(_)) {
+            // todo!(youngsofun): need to refactor parser
+            return Err(ErrorCode::SyntaxException(
+                "copy into table from uri with transform not supported yet",
+            ));
+        }
+
+        let (mut stage_info, path) =
+            parse_file_location(&self.ctx, location, BTreeMap::new()).await?;
+        self.apply_stage_options(stmt, &mut stage_info).await?;
+        let files_info = StageFilesInfo {
+            path,
+            pattern: stmt.pattern.clone(),
+            files: stmt.files.clone(),
+        };
+
+        let start = Instant::now();
+        {
+            let status = "begin to list files";
+            self.ctx.set_status_info(status);
+            info!(status);
+        }
+
+        let operator = init_stage_operator(&stage_info)?;
+        let files = if operator.info().can_blocking() {
```

Comment:
Not sure if the heavy IO in the planning binder is ok or not? cc @leiysky @sundy-li

Reply:
list s3

Reply:
It's already in async runtime. we don't need to use `blocking_list` I think.

Reply:
maybe faster for local fs @RinChanNOWWW ？

Reply:
> Not sure if the heavy IO in the planning binder is ok or not? cc @leiysky @sundy-li

no real harm for now I think.

the only other choice is delay binding and every thin to pipeline building,  which is worse.

Reply:
I think `blocking_list` can let it not yield the CPU and maybe faster for local fs.